api/backend/blueflood/blueflood.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blueflood
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/square/metrics/api"
+)
+
+type Blueflood struct {
+	api      api.API
+	baseUrl  string
+	tenantId string
+}
+
+type QueryResponse struct {
+	Values []MetricPoint `json:"values"`
+}
+
+type MetricPoint struct {
+	Points    int     `json:"numPoints"`
+	Timestamp int64   `json:"timestamp`
+	Average   float64 `json:"average"`
+	Max       float64 `json:"max"`
+	Min       float64 `json:"min"`
+	Variance  float64 `json:"variance"`
+}
+
+func NewBlueflood(api api.API, baseUrl string, tenantId string) *Blueflood {
+	return &Blueflood{api: api, baseUrl: baseUrl, tenantId: tenantId}
+}
+
+func (b *Blueflood) Api() api.API {
+	return b.api
+}
+
+func (b *Blueflood) FetchSeries(metric api.TaggedMetric, predicate api.Predicate, sampleMethod api.SampleMethod, timerange api.Timerange) (*api.SeriesList, error) {
```

Comment:
i think it would be cool if the body of this function is cut into 2~3 helper functions of the form (prepare-request, perform, clean-response)

Example:

```go
request, err := prepareRequest(...)
rawResponse, err := http.get(...)
response, err := processRespones(rawResponse)
```